src/Models/Permission.php

```diff
@@ -137,11 +137,64 @@ public static function findOrCreate(string $name, $guardName = null): Permission
 
     /**
      * Get the current cached permissions.
+     *
+     * @param array $params
+     * @param bool $onlyOne
+     *
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    protected static function getPermissions(array $params = [], bool $onlyOne = false): Collection
+    {
+        return app(PermissionRegistrar::class)->getPermissions($params, $onlyOne);
+    }
+
+    /**
+     * Get the current cached first permission.
+     *
+     * @param array $params
+     *
+     * @return \Spatie\Permission\Contracts\Permission
+     */
+    protected static function getPermission(array $params = []): ?PermissionContract
+    {
+        return static::getPermissions($params, true)->first();
+    }
+
+    /**
+     * Fill model from array.
+     *
+     * @param array $attributes
+     */
+    protected function fillModelFromArray(array $attributes)
+    {
+        if (isset($attributes['roles'])) {
+            $roleClass = app(PermissionRegistrar::class)->getRoleClass();
+            $this->relations['roles'] = new Collection();
+
+            foreach ($attributes['roles'] as $value) {
+                $this->relations['roles']->push($roleClass::getModelFromArray($value));
+            }
+            unset($attributes['roles']);
+        }
+
+        $this->attributes = $attributes;
+        if (isset($attributes['id'])) {
+            $this->exists = true;
+            $this->original['id'] = $attributes['id'];
+        }
+        return $this;
+    }
+
+    /**
+     * Get model from array.
+     *
+     * @param array $attributes
+     *
+     * @return \Spatie\Permission\Contracts\Permission
      */
-    protected static function getPermissions(array $params = []): Collection
+    public static function getModelFromArray(array $attributes): ?PermissionContract
     {
-        return app(PermissionRegistrar::class)
-            ->setPermissionClass(static::class)
```

Comment:
Is removal of this call to `setPermissionClass()` intentional? What's lost by doing this?

Reply:
intentional, because
https://github.com/spatie/laravel-permission/blob/771158773ef29db36ae8949153c7ddfa72bae9a4/src/PermissionRegistrar.php#L45
but thanks for point that, i gonna revert that change, because i didn't read
>This is only intended to be called by the PermissionServiceProvider on boot, so that long-running instances like Swoole don't keep old data in memory.

So, it is an important fix, I suppose